### ISSUE TITLE
ppxlib 5.2 trunk-support needs value_constraint support

### ISSUE THREAD

**Issue by avsm** [ocaml-ppx/ppxlib#423]
Since ocaml/ocaml@ba57d5212a, the trunk-support branch fails to build due to ocaml/ocaml#12188 changing how coercion on value bindings work.

**Comment by pitag-ha** [comment]
I'm working on that. Adding support for value binding coercions was straight-forward, but by doing so, I've noticed that our migrations of value binding constraints already seem to be wrong, so it's taking a bit longer. I can see if I already push something that compiles on 5.1 and 5.2 today, even if it won't do the right migrations. That would also be helpful for @Sudha247 and @shym, wouldn't it?

**Comment by pitag-ha** [comment]
@avsm, @Sudha247, @shym and @nojb, you can pin to [panglesd:fix-418](https://github.com/panglesd/ppxlib/tree/fix-418), if you already need `5.1.0~alpha2` or `5.2.0+trunk` support (now that I've said that, I won't force push to it anymore). We'll probably merge the branch beginning of next week, so then our `trunk-support` will be working again.

**Comment by Sudha247** [comment]
Thanks! cc @punchagan relevant for https://github.com/ocaml-bench/sandmark/issues/453.

**Comment by avsm** [comment]
@pitag-ha your speed of response is faster than light!  Thanks very much indeed -- your branch above unblocked me while testing the ppc64/ocaml5 PR.

**Comment by shym** [comment]
@pitag-ha I had a quick glance at that branch (probably too short to count as proper review, but still), I really like its testing, very nice!
